File: 50 (Pow(x, n))/javascript/Solution.js
/**
 * @param {number} x
 * @param {number} n
 * @return {number}
 */
var myPow = function(x, n) {
  if (n<0){
      n = -n;
      x = 1/x;
  }
  let res= 1.0;
  while (n!=0){
      if (n&1) res*=x;
      x*=x;
      n>>>=1;
  }
  return res;
};

File: 50 (Pow(x, n))/cpp/Solution.cpp
class Solution {
public:
    double myPow(double x, int num) {
        if (num<0){
            x = 1/x;
        }
        long n = labs(num);
        double res= 1.0;
        while (n!=0){
            if (n&1) res*=x;
            x*=x;
            n>>=1;
        }
        return res;
    }
};